ndk-patches/stdio.h.patch

```diff
@@ -10,7 +9,7 @@ diff -uNr ./usr.old/include/stdio.h ./usr/include/stdio.h
 +
  #include <bits/seek_constants.h>
  
--#if __ANDROID_API__ < __ANDROID_API_N__
+-#if __ANDROID_API__ < 24
```

Comment:
Can we try to get rid of this change?

Reply:
The reason I'd like this removed is that I patched some Swift packages to work against the unpatched NDK version of this header, apple/swift-nio-ssl#306 and apple/swift-tools-support-core#243, which I now have to revert to build in Termux.

Whatever the reason this was patched in 11d7363 a couple years ago, we should fix that rather than change the definition of that struct through that header.

Reply:
With or without this part of the patch I get an error about undefined FILE* when building for example bgrep:
```
In file included from argp-pvh.c:23:
In file included from ./argp.h:22:
In file included from ./stdio.h:43:
In file included from /work/termux-build/_cache/android-r23-api-24-v0/bin/../sysroot/usr/include/stdio.h:47:
In file included from ./string.h:52:
In file included from ./stdlib.h:36:
In file included from /work/termux-build/_cache/android-r23-api-24-v0/bin/../sysroot/usr/include/stdlib.h:34:
/work/termux-build/_cache/android-r23-api-24-v0/bin/../sysroot/usr/include/malloc.h:168:37: error: unknown type name 'FILE'
int malloc_info(int __must_be_zero, FILE* __fp) __INTRODUCED_IN(23);
                                    ^
```
might be this error that it was suppose to fix

Reply:
I took a look: as you said, it happens whether you keep this patch or not. Based on the include list you pasted and that I reproduced, this package is doing some weird stuff with custom `stdio.h` and `stdlib.h` files, and it does include the NDK `stdio.h` up to line 47 but then jumps into its string.h _before_ `FILE` is defined, which then breaks in the NDK `malloc.h` when `FILE` is invoked.

Now, that `malloc.h` header does try to include `stdio.h` to get it, but presumably it doesn't because it was trying to load that header in the first place, which would be a circular include. I never bothered to know C's dumb, convoluted include rules, but this package is playing with them and failing unrelated to this patch.

Reply:
@shoya0x00 just FYI, since you are the maintainer for bgrep, the package will be [disabled](https://github.com/termux/termux-packages/pull/7339/commits/62cf147426057761d6a61341eabbcee7cddbbb7e) when we update to ndk r23, unless the package is patched to be build-able with ndk r23